app/report/utils.py

```diff
@@ -44,3 +51,97 @@ def get_csv_file_data(serialized_notifications: list[Any], lang="en") -> bytes:
     string = csv_file.getvalue()
     encoded_string = string.encode("utf-8")
     return encoded_string
+
+
+def build_notifications_query(service_id, notification_type, days_limit=7):
+    """
+    Builds and returns an SQLAlchemy query for notifications with the specified parameters.
+
+    Args:
+        service_id: The ID of the service to query
+        notification_type: The type of notifications to include
+        days_limit: Number of days to look back in history
+
+    Returns:
+        SQLAlchemy query object for notifications
+    """
+    # Create aliases for the tables to make the query more readable
+    n = aliased(Notification)
+    t = aliased(Template)
+    j = aliased(Job)
+    u = aliased(User)
+
+    # Build the query using SQLAlchemy
+    return (
+        db.session.query(
+            n.to.label("Recipient"),
+            t.name.label("Template"),
+            n.notification_type.label("Type"),
+            func.coalesce(u.name, "").label("Sent by"),
```

Comment:
when i ran this locally both the sent_by and sent_by_email were empty in the file

Reply:
I believe that is supposed to happen if the sends were via the api. Did you send via api calls?